volatility/framework/symbols/windows/extensions/__init__.py

```diff
@@ -666,3 +700,81 @@ def to_list(self,
 
     def __iter__(self) -> Iterator[interfaces.objects.ObjectInterface]:
         return self.to_list(self.vol.parent.vol.type_name, self.vol.member_name)
+
+
+class TOKEN(objects.StructType):
+    """A class for process etoken object."""
+
+    def get_sids(self) -> Iterable[str]:
+        """Yield a sid for the current token object."""
+
+        userAndGroupCount = int(self.UserAndGroupCount)
+        if userAndGroupCount < 0xFFFF:
+            layer_name = self.vol.layer_name
+            kvo = self._context.layers[layer_name].config["kernel_virtual_offset"]        
+            symbol_table = self.get_symbol_table_name()
+            ntkrnlmp = self._context.module(symbol_table,
+                                      layer_name = layer_name,
+                                      offset = kvo)
+            UserAndGroups = ntkrnlmp.object(object_type="array",
+                                       offset=self.UserAndGroups.dereference().vol.get("offset") - kvo,
+                                       subtype = ntkrnlmp.get_type("_SID_AND_ATTRIBUTES"),
+                                       count=userAndGroupCount)
+            for saa in UserAndGroups:
+                sid = saa.Sid.dereference().cast("_SID")
+                 # catch invalid pointers (UserAndGroupCount is too high)
+                if sid == None:
+                    raise StopIteration
+                # this mimics the windows API IsValidSid
+                if sid.Revision & 0xF != 1 or sid.SubAuthorityCount > 15:
+                    raise StopIteration
+                id_auth = ""
+                for i in sid.IdentifierAuthority.Value:
+                    id_auth = i
+                SubAuthority = ntkrnlmp.object(object_type="array",
+                                               offset=sid.SubAuthority.vol.offset - kvo,
+                                               subtype = ntkrnlmp.get_type("unsigned long"),
+                                               count= int(sid.SubAuthorityCount))
+                yield "S-" + "-".join(str(i) for i in (sid.Revision, id_auth) +
+                                      tuple(SubAuthority))
+            
+
+    def privileges(self):
+        "Return a list of privileges for the current token object."
+        for priv_index in range(64):
+            yield (priv_index,
+                   self.Privileges.Present & (2**priv_index) !=0,
+                   self.Privileges.Enabled & (2**priv_index) !=0,
+                   self.Privileges.EnabledByDefault & (2**priv_index) !=0)
+
+
+class KTHREAD(objects.StructType):
+    """A class for thread control block objects."""
+        
+    def get_state(self) -> str:
+        dictState = {0:'Initialized', 1: 'Ready', 2: 'Running', 3: 'Standby', 4: 'Terminated',
+                    5: 'Waiting', 6: 'Transition', 7: 'DeferredReady', 8: 'GateWait'}
+        state = int(self.State)
+
+        if state in dictState:
```

Comment:
These three lines could probably be reduced to:
`return dictState.get(state, renderers.NotApplicableValue()`